frontend/packages/console-shared/src/types/pod.ts

```diff
@@ -0,0 +1,42 @@
+import { ObjectMetadata } from '@console/internal/module/k8s';
+
+export interface Pod {
+  id?: string;
+  name?: string;
+  kind?: string;
+  metadata?: {};
+  status: { phase: string };
+  spec: {};
+}
+
+export interface ResourceProps {
+  kind: string;
+  apiVersion?: string;
+  metadata: ObjectMetadata;
+  status: {};
+  spec: {
+    [key: string]: any;
+  };
+}
+
+export interface Resource {
+  data: ResourceProps[];
+}
+
+export interface TopologyDataResources {
+  replicationControllers: Resource;
+  pods: Resource;
+  deploymentConfigs: Resource;
+  services: Resource;
+  routes: Resource;
+  deployments: Resource;
+  replicasets: Resource;
+  buildconfigs: Resource;
+  builds: Resource;
+  daemonSets?: Resource;
+  ksroutes?: Resource;
+  configurations?: Resource;
+  revisions?: Resource;
```

Comment:
Why topology types in pod status?
Also seems like there are types that aren't needed. eg `revisions`